### PR TITLE
External data connector REST methods

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -2525,6 +2530,42 @@ object will read the local file instead.
       'timeseries/51922d0b37203f2a8c000010', connection);
 ```
 
+External Connectors
+-------------------
+
+BigML offers an `externalconnector` resource that can be used to connect to
+external data sources. The description of the API requirements to create an
+`ExternalConnector` can be found at the
```
Comment (sdesimone) [inline docs/index.md:2538]: in the api...

File: docs/index.md [hunk 2]
```diff
@@ -2525,6 +2530,42 @@ object will read the local file instead.
...
+[API documentation](https://bigml.com/api/externalconnectors). The compulsory
```
Comment (sdesimone) [inline docs/index.md:2539]: "Required" to replace "The compulsory"

File: docs/index.md [hunk 3]
```diff
@@ -2525,6 +2530,42 @@ object will read the local file instead.
...
+export BIGML_EXTERNAL_CONN_SOURCE="postgresql"
+```
+
+
+```js
+    var bigml = require('bigml');
+    var externalConnectorConn = new bigml.ExternalConnector(),
+      externalConnectorId, connectionInfo,
+      args = {"name": "my connector"}, retry = false;
+    externalConnectorConn.create(connectionInfo, args, retry,
```
Comment (sdesimone) [inline docs/index.md:2560]: I think it would be useful here to show how you inutilizare `connectionInfo`

Comment (mmerce) [reply docs/index.md:2560]: inutilizare?

Comment (mmerce) [reply docs/index.md:2560]: The connectionInfo in this case is set to undefined but internally the class retrieves the info in the environment variables.

Comment (sdesimone) [reply docs/index.md:2560]: that should have been initialise -- don't know how it morphed into "inutilizare"...

File: docs/index.md [hunk 4]
```diff
@@ -334,7 +335,11 @@ through this `LocalEnsemble` object.
...
 ------------------
 
-Currently there are six types of resources in bigml.com:
+Currently these are the types of resources in bigml.com:
+
+- **external connectors** Contain the information to connect to an external
+database manager. They can be used to upload data to BigML to build `sources`.
+These resources are handled through `bigml.ExternalConnector`.
 
 - **sources** Contain the data uploaded from your local data file after
```
Comment (sdesimone) [inline docs/index.md:344]: also add external sources here : sources** Contain the data uploaded from your local data file or retrieved from an external connector

File: docs/index.md [hunk 5]
```diff
@@ -2525,6 +2530,51 @@ object will read the local file instead.
...
+-------------------
+
+BigML offers an `externalconnector` resource that can be used to connect to
+external data sources. The description of the API requirements to create an
+`ExternalConnector` can be found in the
+[API documentation](https://bigml.com/api/externalconnectors). The required
+information to create an external connector are parameters like the type
+of database manager, the host, user, password and table that we need to access.
+This must be provided as first argument to the `bigml.ExternalConnector`
+create method or can be set in environment variables:
```
Comment (sdesimone) [inline docs/index.md:2543]: you could make this more explicit saying: or can be set in environment variables, in which case you pass the first argument as undefined.

**Pull request merged**